packages/block-editor/src/store/reducer.js

```diff
@@ -858,16 +859,26 @@ export const blocks = pipe(
 					if ( updatedAttributeEntries.length === 0 ) {
 						continue;
 					}
-					const newAttributes = {
-						...state.get( clientId ),
-					};
+					hasChange = true;
+					let hasUpdatedAttributes = false;
+					const existingAttributes = state.get( clientId );
+					const newAttributes = {};
 					updatedAttributeEntries.forEach( ( [ key, value ] ) => {
-						newAttributes[ key ] = value;
+						if ( existingAttributes[ key ] !== value ) {
+							hasUpdatedAttributes = true;
+							newAttributes[ key ] = value;
+						}
 					} );
-					newState.set( clientId, newAttributes );
+					hasChange = hasChange || hasUpdatedAttributes;
```

Comment:
This is always going to be true with `hasChange`, no?

Reply:
Oh I see, I forgot to remove something :P

Reply:
should be fixed.